wgpu-core/src/device/resource.rs

```diff
@@ -593,6 +631,18 @@ impl Device {
         }
     }
 
+    /// Stop tracing and return the trace object.
+    ///
+    /// This is mostly useful for in-memory traces.
+    pub fn take_trace(&self) -> Option<Box<dyn trace::Trace + Send + Sync + 'static>> {
```

Comment:
This doesn't work, because `Trace` isn't even defined when the feature isn't enabled.

There is a comment on `wgt::Trace`:
https://github.com/gfx-rs/wgpu/blob/0810110bd763ddbb0794bafd72679a895692bb3b/wgpu-types/src/lib.rs#L1514

I'm not sure I fully understand the concern here. Would it be a problem if `take_trace` is only available when the feature is enabled?